lib/wickedbase.pm

```diff
@@ -1306,8 +1306,8 @@ sub wait_for_background_process {
     my ($self, $pid, %args) = @_;
     $args{proceed_on_failure} //= 0;
 
-    my $ret = script_run("wait $pid", die_on_timeout => 0, %args);
-    unless (defined($ret)) {
+    my $ret = script_run("waitpid --timeout 25 $pid", %args);
+    if ($ret == 3) {    # timeout
```

Comment:
This causes a regression in older systems where `utils-linux` doesn't have `waitpid`. (see: https://openqa.suse.de/tests/15183353#step/t11_DAD_auto4_detect/87)

Wait pid was introduced with
```
 Tue Jan 23 2024 sbrabec@suse.com
- Upgrade to version 2.39.3 (PED-7694):
```

As the command `timeout` is not usable with bash internal functions. Maybe this could be a workaround?
```
my $ret = eval {
  script_run("wait $pid", %args);
};
unless (defined($ret)) {
```
Does someone see a better way?

We need to cleanup this console, because otherwise the `post-run` script do not run and we collect all important information there, especially in error case these infos are important.

FYI @rfan1

Reply:
> We need to cleanup this console, because otherwise the post-run script do not run and we collect all important information there, especially in error case these infos are important.

Can you use a different console instead?

Reply:
I wonder whether this caused https://openqa.opensuse.org/tests/4405117. It looks like two jobs are waiting for each other, deadlocked.

Reply:
> I wonder whether this caused https://openqa.opensuse.org/tests/4405117. It looks like two jobs are waiting for each other, **deadlocked.

It is related. But I think the core problem is different.
here we see 
```
# waitpid --timeout 25 10288; echo n_hPH-$?-
waitpid: could not open pid 10288: No such process
n_hPH-1-
```
I guess waitpid has a different behavior then inline function wait here. 
And it looks like `wait` exited with 0 (if the process was already finished) so the test didn't died before.

anyway @rfan1 is working on a solution https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19981

Reply:
The VRs for both sle and tw passed:

[sle](http://openqa.suse.de/tests/15199237#dependencies)
[tw](https://openqa.opensuse.org/tests/4407304#dependencies)

Reply:
> Can you use a different console instead?
it is, but not really user friendly (in my opinion). Take a look into https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/a27ead9722008d3817803916a22cc686e098a7c8/tests/wicked/basic/ref/t11_DAD_auto4_detect.pm#L40

The things we do are:
```
1) Start tracer
2) toggle ifc state with wicked 
3) check if tracer finished -- the good case
4) validate the output of the tracer
5) Notify parallel job to be ready
```

I guess a other solution would be to give the tracer (which is a python script using scapy) the ability to have a timeout.